### PR TITLE
Sophia - add Rider Create Page

### LINKED ISSUE #39: add Rider Create Page
Add a Rider Create Page to meet acceptance criteria for #7

### PR DESCRIPTION
**Pull request by SophiaWeiler**
In this PR, I have created a new page to create a new ride request called RiderCreatePage. It also has tests and a [storybook](https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-4/prs/36/storybook/?path=/story/components-nav-appnavbar--no-role) component (which I have included a small screen shot below to document what the page will look like). Makes progress on #7 

Screen Shot 2023-06-01 at 3 13 46 PM

### CODE REVIEW

**Review by zachmiller-ucsb — DISMISSED** [review]
LGTM - checked code quality + storybook entry

**Review by mike-k999 — DISMISSED** [review]
lgtm

File: frontend/src/main/pages/Rider/RiderCreatePage.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+import RiderForm from "main/components/Rider/RiderForm";
+import { Navigate } from 'react-router-dom'
+import { useBackendMutation } from "main/utils/useBackend";
+import { toast } from "react-toastify";
+import { useCurrentUser } from "main/utils/currentUser";
+
+export default function RiderCreatePage() {
```
Comment (pconrad) [inline frontend/src/main/pages/Rider/RiderCreatePage.js]: Can we call this a name that matches what the user see and thinks of this as?
```suggestion
export default function RideRequestCreatePage() {
```

I've had a lot of experience with code bases where names in the code are out of alignment with what the  user thinks and sees. Everyone hates working in those code bases.  The best time to fix it is before it goes into the main branch.

File: frontend/src/stories/pages/Rider/RiderCreatePage.stories.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+import RiderCreatePage from 'main/pages/Rider/RiderCreatePage';
```
Comment (pconrad) [inline frontend/src/stories/pages/Rider/RiderCreatePage.stories.js]:
```suggestion
import RideRequestCreatePage from 'main/pages/RideRequest/RideRequestCreatePage';
```

File: frontend/src/stories/pages/Rider/RiderCreatePage.stories.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export default {
+    title: 'pages/Rider/RiderCreatePage',
```
Comment (pconrad) [inline frontend/src/stories/pages/Rider/RiderCreatePage.stories.js]: You can guess what I want here

File: frontend/src/tests/pages/Rider/RiderCreatePage.test.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
+import { render, waitFor, fireEvent } from "@testing-library/react";
```
Comment (pconrad) [inline frontend/src/tests/pages/Rider/RiderCreatePage.test.js]: Please do an incremental global search / replace one at a time on `Rider` and see for each one, whether it should be `RideRequest`.

Also: what's in the backend?  Is it Rider?  or RideRequest?  Do we need to fix it there also?

**Review by pconrad — CHANGES_REQUESTED** [review]
naming, naming, naming....

**Review by mike-k999 — COMMENTED** [review]
Looks like zach changed "rider" --> "rideRequest", while in the previous PR I had changed "rider" --> "ride".  This is most likely the merge conflict and personally I feel just using "ride" would be shorter and simpler.

**Review by strongani — APPROVED** [review]
LGTM!

**Review by mike-k999 — APPROVED** [review]
Ok so all pages are "RideRequest" and form is just "Ride".  We will remember this convention in this case.  LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**